src/serdes/binary-parser.ts

```diff
@@ -0,0 +1,98 @@
+import { strict as assert } from 'assert'
+import { makeClass } from '../utils/make-class'
+import { Enums } from '../enums'
+import { slice, parseBytes } from '../utils/bytes-utils'
+
+const BinaryParser = makeClass({
+  BinaryParser (buf) {
+    this._buf = parseBytes(buf, Uint8Array)
+    this._length = this._buf.length
+    this._cursor = 0
+  },
+  skip (n) {
+    this._cursor += n
+  },
+  read (n, to = Uint8Array) {
+    const start = this._cursor
+    const end = this._cursor + n
+    assert(end <= this._buf.length)
+    this._cursor = end
+    return slice(this._buf, start, end, to)
+  },
+  readUIntN (n) {
+    return this.read(n, Array).reduce((a, b) => a << 8 | b) >>> 0
+  },
+  readUInt8 () {
+    return this._buf[this._cursor++]
+  },
+  readUInt16 () {
+    return this.readUIntN(2)
+  },
+  readUInt32 () {
+    return this.readUIntN(4)
+  },
+  pos () {
+    return this._cursor
+  },
+  size () {
+    return this._buf.length
+  },
+  end (customEnd) {
+    const cursor = this.pos()
+    return (cursor >= this._length) || (customEnd !== null &&
+            cursor >= customEnd)
+  },
+  readVL () {
+    return this.read(this.readVLLength())
+  },
+  readVLLength () {
+    const b1 = this.readUInt8()
+    if (b1 <= 192) {
+      return b1
+    } else if (b1 <= 240) {
+      const b2 = this.readUInt8()
+      return 193 + (b1 - 193) * 256 + b2
```

Comment:
Similarly, in the future it would be great if these were not magic numbers? What does 193 represent?